### ISSUE TITLE
Privacy Pass Protocol support

### ISSUE THREAD

**Issue by Destroyinator69420** [OnionBrowser/iOS-OnionBrowser#337]
Privacy Pass is a protocol supported by Cloudflare Captcha and Hcaptcha that allows users to solve one Captcha and earn privacy passes so they do not have to solve the next one. The privacy pass devs can be found at https://privacypass.github.io/. The source code for the extension can be found at https://github.com/privacypass/challenge-bypass-extension, and the protocol is documented at https://privacypass.github.io/protocol/.

All the best,
Destroyinator69420

**Comment by tladesignz** [comment]
Yeah, we already thought about this. Unfortunately, Mobile Safari cannot load plugins, so that would mean a complete rewrite in Swift. We'll see, what iOS 15 will bring in that regard. Apple announced, that Mobile Safari will support Webextension Plugins. Not sure if that support will end up in `WKWebView`, too. Where we still need to migrate to...

**Comment by tladesignz** [comment]
Huh... There's nows on that front, @n8fr8:

https://blog.cloudflare.com/how-to-enable-private-access-tokens-in-ios-16-and-stop-seeing-captchas/

https://developer.apple.com/news/?id=huqjyh7k

Maybe we don't even need to do anything to support it with OB3?

**Comment by n8fr8** [comment]
It would be great to have support here, whether we have to write it ourselve or not.

**Comment by 404b** [comment]
Privacy pass barely works for you and collects lot of personal and browsing data.

Try **noCaptcha Ai Chrome Extension** Free 6000 req/month - 200/day. Roughly you can solve 70 hcaptcha/daily with free plan, which is mostly sufficient for personal use.

 Chrome Extension Source Here: https://github.com/noCaptchaAi/chrome-extension
